### DISCUSSION TITLE
Pleaseeeeee help

### DISCUSSION THREAD

**Discussion by UjjwalAnand364** [raysan5/raylib#3659] [Q&A] [answered]
Its killed a lot of time and its now killing my interest. 
I first tried installing from source, and I followed the [video](https://www.youtube.com/watch?v=HPDLTQ4J_zQ), but I got the following error, 
```
Makefile:694: recipe for target 'raudio.o' failed
mingw32-make: *** [raudio.o] Error 1
```
and the operation didn't complete as I found no libraylib.a file.

Then, I tried using the installer, and got the libraylib.a file after completion of the installation process.

But, I am unable to run any example file, even those that have been given in the raylib folder, and get the following error:
```
C:\Users\dell\AppData\Local\Temp\ccPMu8hY.o:main.cpp:(.text.startup+0x60): undefined reference to `InitWindow'
C:\Users\dell\AppData\Local\Temp\ccPMu8hY.o:main.cpp:(.text.startup+0x6c): undefined reference to `SetTargetFPS'
C:\Users\dell\AppData\Local\Temp\ccPMu8hY.o:main.cpp:(.text.startup+0x81): undefined reference to `BeginDrawing'
C:\Users\dell\AppData\Local\Temp\ccPMu8hY.o:main.cpp:(.text.startup+0x8f): undefined reference to `ClearBackground'
C:\Users\dell\AppData\Local\Temp\ccPMu8hY.o:main.cpp:(.text.startup+0xcf): undefined reference to `DrawCircle'
C:\Users\dell\AppData\Local\Temp\ccPMu8hY.o:main.cpp:(.text.startup+0xd4): undefined reference to `EndDrawing'
C:\Users\dell\AppData\Local\Temp\ccPMu8hY.o:main.cpp:(.text.startup+0xd9): undefined reference to `WindowShouldClose'
C:\Users\dell\AppData\Local\Temp\ccPMu8hY.o:main.cpp:(.text.startup+0xe2): undefined reference to `CloseWindow'
collect2.exe: error: ld returned 1 exit status
```
The command that I used last is:
`g++ main.cpp -o game.exe -O2 -Wall -Wno-missing-braces -L lib/ -I include/ -lraylib -lopengl32 -lgdi32 -lwinmm`

I have looked [here](https://github.com/raysan5/raylib/discussions/2210) but to no avail. The compiler is unable to find the functions, and I reckon some sort of command or installing issue.
HELP!

**Comment by 1NEGROUP** [comment]
are you using Visual studio code?

**Reply by UjjwalAnand364** [reply]
Yes I am using VS code. But has that got to do something with it?

**Comment by Peter0x44** [comment]
@UjjwalAnand364 I suspect I know the reason (if gcc --version mentions mingw.org, that's why).
It would be useful if you joined the raylib discord server and people could help you in a more real time manner.

**Reply by UjjwalAnand364** [reply]
Thanks, I'll check out the discord. 
Btw, I ran gcc --version and the output is:
`gcc.exe (x86_64-posix-seh-rev1, Built by MinGW-Builds project) 13.2.0`

**Comment by 1NEGROUP** [comment]
> Yes I am using VS code. But has that got to do something with it?

Yeah It Might I had Issues with vs code and if your using linux then i would suggest not using it. I installed vs codium and that worked for me but i can see that your using windows so i would try that and I would Look for your include files if your using g++ then your include files should be in C:\msys64\mingw64 you should see the include and libs folder

**Comment by UjjwalAnand364** [comment, marked as answer]
I did get it going now, and here's a tip to anyone new out there trying this. 
The videos and other sources mostly don't mention the environment settings too clearly, like the mingw and gcc version, so try upgrading it to the latest one. 
I think the mingw and gcc version and other details like adding gcc to path should be mentioned on the 'Working with Windows' wiki page, so newbies don't go crazy trying to solve their problem.
